qendpoint-store/src/main/java/com/the_qa_company/qendpoint/compiler/SparqlRepository.java

```diff
@@ -541,6 +519,37 @@ private ClosableResult<?> execute0(RepositoryConnection customConnection, String
 
 			ParsedQuery parsedQuery = QueryParserUtil.parseQuery(QueryLanguage.SPARQL, sparqlQuery, null);
 
+			// Substitute [AUTO_LANGUAGE] with the client's language
+			if (acceptLanguageHeader != null) {
+				List<Locale.LanguageRange> languageRanges = Locale.LanguageRange.parse(acceptLanguageHeader);
+				Locale locale = Locale.lookup(languageRanges, Arrays.asList(Locale.getAvailableLocales()));
+				try {
+					parsedQuery.getTupleExpr().visit(new AbstractQueryModelVisitor<Exception>() {
+						@Override
+						public void meet(Var node) throws Exception {
+							if (node.getValue() instanceof Literal literal) {
+								boolean isStringDatatype = literal.getDatatype() == null
+										|| literal.getDatatype().equals(CoreDatatype.XSD.STRING.getIri());
```

Comment:
- you can use [`isLiteral`](https://rdf4j.org/javadoc/latest/org/eclipse/rdf4j/model/Literal.html#isLiteral()) instead of instanceof, it is faster
- using [`getCoreDatatype()`](https://rdf4j.org/javadoc/latest/org/eclipse/rdf4j/model/Literal.html#getCoreDatatype()) is better than using the datatype, using that you avoid comparing the datatype string, so it's faster.
- why not inlining the isStringDatatype?